query-languages/m/type-is.md

```diff
@@ -12,7 +12,9 @@ Type.Is(<b>type1</b> as type, <b>type2</b> as type) as logical
 
 ## About
 
-Determines if a value of `type1` is always compatible with `type2`.
+Determines if a value of `type1` is always compatible with `type2`. 
+
+Parameter `type2` is expected to be a nullable primitive type. When this requirement is not met, this function's behavior is undefined and so should not be relied on.
```

Comment:
This is not true. By passing a nullable primitive type as the second arg, you can use Type.Is to check for both the nullable and non-nullable version of that type. But that's just one use of the function.

Here's another way it can be used, which doesn't involve nullable types at all:
Type.Is(Int64.Type, type number)

Reply:
Hi @ehrenMSFT,

Thanks for this feedback. 

> Here's another way it can be used, which doesn't involve nullable types at all:
Type.Is(Int64.Type, type number)

I'm puzzled. Isn't `type number` a nullable primitive type (where nullable primitive type = all primitive types + their nullable counterparts)? 

In the [language spec](https://learn.microsoft.com/en-us/powerquery-m/m-spec-types), operators `as` and `is` are described as only accepting "nullable primitive types as their right operand," and `Value.Is`'s second argument is described as accepting "an arbitrary type value as its first and a nullable primitive type value as its second argument." Examples are given like `Type.Is(type number, type text)` and `1 is number`.

Reply:
Not sure about the spec. I'm just basing my comments on how things actually seem to work.

Type.IsNullable(type number) returns false. You'd have to say "type nullable number" to make it nullable.

Reply:
Thanks, @ehrenMSFT!

Just opened a [PR](https://github.com/MicrosoftDocs/query-docs/pull/203) raising the question of how to best address the "nullable primitive type" terminology used in the spec.